fmt/src/selection.rs

```diff
@@ -182,15 +180,45 @@ fn select_files_with_git(
         builder.build().context(SelectWithIgnoreSnafu)?
     };
 
-    let mut it = WalkDir::new(basedir).into_iter();
+    let basedir = basedir
+        .canonicalize()
+        .with_context(|_| ResolveAbsolutePathSnafu {
+            path: basedir.display().to_string(),
+        })?;
+    let mut it = WalkDir::new(basedir).follow_links(false).into_iter();
+
+    let workdir = repo
+        .work_dir()
+        .context(GixPathNotFountSnafu { path: "workdir" })?;
+    let workdir = workdir
+        .canonicalize()
+        .with_context(|_| ResolveAbsolutePathSnafu {
+            path: workdir.display().to_string(),
+        })?;
+    let worktree = repo
+        .worktree()
+        .context(GixPathNotFountSnafu { path: "worktree" })?;
+    let mut excludes = worktree
+        .excludes(None)
```

Comment:
Note that this will fail if there is no index file, which typically happens only in bare or newly initialized repos. The latter could definitely fail here. If that's a problem, one would want to use `repo.excludes(&repo.index_or_empty()?, …)`.

Reply:
Marked. Let's leave it as you wrote now. It's not quite a frequent case.